controllers/usersignup/usersignup_controller.go

```diff
@@ -429,6 +427,38 @@ func (r *Reconciler) ensureNewMurIfApproved(reqLogger logr.Logger, config toolch
 	return r.provisionMasterUserRecord(reqLogger, config, userSignup, targetCluster, userTier)
 }
 
+func (r *Reconciler) getUserTier(reqLogger logr.Logger, config toolchainconfig.ToolchainConfig, userSignup *toolchainv1alpha1.UserSignup) (*toolchainv1alpha1.UserTier, error) {
+	tierName := config.Tiers().DefaultUserTier()
+	if eventName, found := userSignup.Labels[toolchainv1alpha1.SocialEventUserSignupLabelKey]; found {
+		event := &toolchainv1alpha1.SocialEvent{}
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: userSignup.Namespace, Name: eventName}, event); err != nil && !errors.IsNotFound(err) {
+			return nil, err
+		} else if err == nil {
+			tierName = event.Spec.UserTier
+		}
+	}
```

Comment:
This could be extracted into a function 
```
getSocalEvent(userSignup *toolchainv1alpha1.UserSignup) (bool, SocialEvent)
```
so the implementation and lookup for the label is at one place

Reply:
> > BTW, it looks like we are also missing a logic for `SocialEvent.Spec.PreferSameCluster` parameter - do we have a jira for this?
> 
> good point, let me check that

https://issues.redhat.com/browse/CRT-1616

Reply:
done in d76e48e